config/v1/types_ingress.go

```diff
@@ -131,6 +131,7 @@ type LoadBalancer struct {
 
 // AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
 // This only includes fields that can be modified in the cluster.
+// +kubebuilder:validation:XValidation:rule="self.type != 'Classic' || !has(self.networkLoadBalancer)",message="Network load balancer parameters are allowed only when load balancer type is NLB."
```

Comment:
You should revise this, since currently it is not expandable if we create new values in the enum in the future, try something like
```suggestion
// +kubebuilder:validation:XValidation:rule="self.type == 'NLB' ? true : !has(self.networkLoadBalancer)",message="Network load balancer parameters are allowed only when load balancer type is NLB."
```
Using a ternary here would be what I'd suggest

Reply:
@JoelSpeed OK what do you mean by "since currently it is not expandable if we create new values in the enum in the future" ? Just asking for learning purpose so that I can keep this in mind.

Reply:
Your current logic says, I cannot use networkLoadBalancer if the value of type is Classic. But, imagine I added a new value to the type enum, ALB, now, according to your current validation, I can use the `networkLoadBalancer` with the ALB type.